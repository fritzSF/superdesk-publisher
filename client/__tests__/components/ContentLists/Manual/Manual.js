import React from "react";
import Manual from "../../../../components/ContentLists/Manual/Manual";
import { render, waitForElement } from "@testing-library/react";
import axios from "axios";

import Publisher from "../../../../__mocks__/publisher";

const publisher = new Publisher();

let lists = [];
publisher.queryLists().then(items => (lists = items));

jest.mock("axios");
jest.mock("react-select", () => props => "div");

let api = () => {};

api.users = {};
api.users.query = function() {
  return new Promise((resolve, reject) => {
    resolve({
      _items: [{ is_author: true, display_name: "test author" }],
      _links: {}
    });
  });
};

describe("ContentLists/Manual/Manual", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <Manual
        list={lists[0]}
        lists={lists}
        publisher={publisher}
        listEdit={jest.fn()}
        onListUpdate={jest.fn()}
        api={api}
        toggleFilters={jest.fn()}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
