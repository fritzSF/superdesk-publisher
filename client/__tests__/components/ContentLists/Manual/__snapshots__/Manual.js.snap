// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLists/Manual/Manual renders properly 1`] = `
<div
  class="flex-grid flex-grid--grow flex-grid--small-2"
>
  <div
    class="flex-grid__item flex-grid__item--d-flex flex-grid__item--column panel-border-right"
  >
    <div
      class="subnav subnav--lower-z-index subnav--dark-blue-grey"
    >
      <button
        class="navbtn navbtn--left"
      >
        <i
          class="icon-th"
        />
      </button>
      <div
        class="flat-searchbar"
        style="background-color: transparent;"
      >
        <div
          class="search-handler"
        >
          <label
            class="trigger-icon"
            for="search-input"
          >
            <i
              class="icon-search"
              style="color: rgb(255, 255, 255);"
            />
          </label>
          <input
            placeholder="Search"
            style="color: rgb(255, 255, 255);"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dropdown undefined"
      >
        <span>
          <button
            class="dropdown__toggle navbtn navbtn--text-only dropdown-toggle"
          >
            Most shared
            <span
              class="dropdown__caret"
            />
          </button>
        </span>
        <ul
          class="dropdown__menu dropdown__menu--scrollable"
        >
          <li>
            <div
              class="dropdown__menu-label"
            />
          </li>
          <li>
            <button>
              Most shared
            </button>
          </li>
          <li>
            <button>
              Top news
            </button>
          </li>
        </ul>
      </div>
      <div
        class="subnav__stretch-bar"
      />
      <span
        class="margin--right"
      >
        <button
          aria-label="button"
          class="btn btn--primary btn--disabled"
        >
          Save
        </button>
      </span>
    </div>
    <div
      class="sd-column-box--3"
    >
      <div
        class="sd-column-box__main-column relative dropZone dropZone--empty"
      >
        <h2
          class="dropZone__heading"
        >
          Drag your Articles here
        </h2>
        <ul
          class="sd-list-item-group sd-shadow--z2"
        />
      </div>
    </div>
  </div>
  <div
    class="flex-grid__item flex-grid__item--d-flex flex-grid__item--column"
  >
    <div
      class="subnav subnav--lower-z-index"
    >
      <button
        class="navbtn navbtn--left navbtn--darker"
        flow="right"
        sd-tooltip="Filter"
      >
        <i
          class="icon-filter-large"
        />
      </button>
      <div
        class="flat-searchbar"
      >
        <div
          class="search-handler"
        >
          <label
            class="trigger-icon"
            for="search-input"
          >
            <i
              class="icon-search"
            />
          </label>
          <input
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
      <h3
        class="subnav__page-title"
      >
        All published articles
      </h3>
      <div
        class="subnav__spacer subnav__spacer--no-margin"
      />
      <div
        class="subnav__content-bar sd-flex-no-shrink sd-margin-x--0"
      >
        <div
          class="dropdown dropdown--align-right"
        >
          <span>
            <button
              class="dropdown__toggle navbtn navbtn--text-only dropdown-toggle"
            >
              All Languages
              <span
                class="dropdown__caret"
              />
            </button>
          </span>
          <ul
            class="dropdown__menu dropdown__menu--scrollable"
          >
            <li>
              <button>
                All Languages
              </button>
            </li>
            <li
              class="dropdown__menu-divider"
            />
            <li>
              <button>
                Polish
              </button>
            </li>
            <li>
              <button>
                English
              </button>
            </li>
            <li>
              <button>
                German
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sd-column-box--3"
    >
      <div
        class="sd-column-box__main-column relative"
      >
        <ul
          class="sd-list-item-group sd-shadow--z2"
        >
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="draggable_1317"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="draggable_1317"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="sd-list-item"
              style="cursor: grab; opacity: 1;"
            >
              <div
                class="sd-list-item__column sd-list-item__column--grow"
              >
                <div
                  class="sd-list-item__row"
                >
                  <span
                    class="sd-overflow-ellipsis sd-list-item__text-strong"
                  >
                    If you let any one have it but me, Basil, I shall never forgive you
                  </span>
                </div>
                <div
                  class="sd-list-item__row"
                >
                  <span
                    class="sd-overflow-ellipsis sd-list-item--element-grow"
                  >
                    2 days ago
                  </span>
                  <span
                    class="label label--success label--hollow"
                  >
                    sport
                  </span>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="sd-filters-panel sd-filters-panel--border-left"
      >
        <div
          class="side-panel side-panel--transparent side-panel--shadow-right"
        >
          <div
            class="side-panel__header side-panel__header--border-b"
          >
            <span
              class="side-panel__close"
            >
              <a
                class="icn-btn"
                data-flow="left"
                data-sd-tooltip="Close"
              >
                <i
                  class="icon-close-small"
                />
              </a>
            </span>
            <h3
              class="side-panel__heading"
            >
              Advanced filters
            </h3>
          </div>
          <div
            class="side-panel__content"
          >
            <div
              class="side-panel__content-block"
            >
              <div
                class="form__row"
              >
                <div
                  class="sd-line-input sd-line-input--no-margin sd-line-input--with-button"
                >
                  <label
                    class="sd-line-input__label"
                  >
                    Categories
                  </label>
                  div
                </div>
              </div>
              <div
                class="form__row"
              >
                <div
                  class="sd-line-input sd-line-input--no-margin sd-line-input--with-button"
                >
                  <label
                    class="sd-line-input__label"
                  >
                    Author
                  </label>
                  div
                </div>
              </div>
              <div
                class="form__row form__row--flex"
              >
                <div
                  class="sd-line-input sd-line-input--no-margin"
                >
                  <label
                    class="sd-line-input__label"
                  >
                    Published date
                  </label>
                  <input
                    class="sd-line-input__input"
                    name="published_at"
                    type="date"
                    value=""
                  />
                </div>
              </div>
              <div
                class="form__row form__row--flex"
              >
                <div
                  class="sd-line-input sd-line-input--no-margin"
                >
                  <label
                    class="sd-line-input__label"
                  >
                    Published after
                  </label>
                  <input
                    class="sd-line-input__input"
                    name="published_after"
                    type="date"
                    value=""
                  />
                </div>
              </div>
              <div
                class="form__row form__row--flex"
              >
                <div
                  class="sd-line-input sd-line-input--no-margin"
                >
                  <label
                    class="sd-line-input__label"
                  >
                    Published before
                  </label>
                  <input
                    class="sd-line-input__input"
                    name="published_before"
                    type="date"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="side-panel__footer side-panel__footer--button-box"
          >
            <div
              class="flex-grid flex-grid--boxed-small flex-grid--small-2"
            >
              <button
                aria-label="button"
                class="btn btn--hollow"
              >
                Clear
              </button>
              <button
                aria-label="button"
                class="btn btn--primary"
              >
                Filter
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
